src/amo/components/InstallButtonWrapper/index.js

```diff
@@ -0,0 +1,128 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import GetFirefoxButton from 'amo/components/GetFirefoxButton';
+import AMInstallButton from 'core/components/AMInstallButton';
+import { UNKNOWN } from 'core/constants';
+import { getVersionById } from 'core/reducers/versions';
+import { getClientCompatibility, isFirefox } from 'core/utils/compatibility';
+import type { GetFirefoxButtonTypeType } from 'amo/components/GetFirefoxButton';
+import type { WithInstallHelpersInjectedProps } from 'core/installAddon';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { AddonVersionType } from 'core/reducers/versions';
+import type { AddonType } from 'core/types/addons';
+import type { AppState } from 'disco/store';
+
+export type Props = {|
+  _getClientCompatibility?: typeof getClientCompatibility,
+  addon: AddonType,
+  className: string,
+  defaultButtonText: string,
+  defaultInstallSource: string,
+  getFirefoxButtonType: GetFirefoxButtonTypeType,
+  puffy?: boolean,
+  ...WithInstallHelpersInjectedProps,
+|};
+
+type InternalProps = {|
+  ...Props,
+  clientApp: string,
+  currentVersion: AddonVersionType | null,
+  error: string | void,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+  userAgentInfo: UserAgentInfoType,
+|};
+
+export const InstallButtonWrapperBase = (props: InternalProps) => {
+  const {
+    _getClientCompatibility = getClientCompatibility,
+    addon,
+    className,
+    clientApp,
+    currentVersion,
+    defaultButtonText,
+    defaultInstallSource,
+    enable,
+    getFirefoxButtonType,
+    hasAddonManager,
+    isAddonEnabled,
+    install,
+    installStatus,
+    installTheme,
+    setCurrentStatus,
+    puffy,
+    uninstall,
+    userAgentInfo,
+  } = props;
+
+  let isCompatible = false;
+  let compatibility;
+
+  if (addon && isFirefox({ userAgentInfo })) {
+    compatibility = _getClientCompatibility({
+      addon,
+      currentVersion,
+      clientApp,
+      userAgentInfo,
+    });
+    isCompatible = compatibility.compatible;
+  }
+
+  return (
+    <div className="InstallButtonWrapper">
```

Comment:
right now the guides page's buttons look a little off (compare to dev):

1. on non-firefox clients, the "only with firefox..." buttons looks like this:

![Alt text](https://monosnap.com/image/v6LdFHnufgcSr2EVLneDzglI2xJIt7.png)

is it possible to add an additional class here (maybe ~ `InstallButtonWrapper--GetFirefoxButton`) so it can be styled in a similar way. There might be a way to style without this but I think it might make this easier..

2. on firefox, the "+ Addons" buttons are no longer vertically centered

Reply:
Thanks @rebmullin. I've looked and I cannot quite figure out exactly what the problem is that's causing #1, above, nor do I exactly understand what you're suggesting about an additional class. Could you be more explicit? I was hoping that the `className` prop for `InstallButtonWrapper`, which allows a caller to add a class to both the `AMInstallButton` and the `GetFirefoxButton` would give us what we need for styling, but maybe that's not good enough.

As for #2, above, I think I've fixed it by adding some styles to `InstallButtonWrapper`, but it could be what I've done will break something else. Could you please take a look after I push the next commit?

Reply:
regarding the first issue. for example I have this and I think this works at all sizes. **But** it will mess up the "+Add" styles a bit so we only want to target the "Get Firefox" button styles.

```diff
diff --git a/src/amo/components/GuidesAddonCard/styles.scss b/src/amo/components/GuidesAddonCard/styles.scss
index dce264300..96504aeec 100644
--- a/src/amo/components/GuidesAddonCard/styles.scss
+++ b/src/amo/components/GuidesAddonCard/styles.scss
@@ -59,14 +59,18 @@
     flex-wrap: nowrap;
   }
 
+  .InstallButtonWrapper {
+    width: 100%; // this is only needed only on non-firefox btn styles
+
+    @include respond-to(medium) {
+      flex: 0 0 180px; // this is only needed only on non-firefox btn styles
+    }
+  }
+
   .GetFirefoxButton {
     font-size: $font-size-default;
     text-align: center;
     white-space: normal;
-
-    @include respond-to(medium) {
-      flex: 0 0 180px;
-    }
   }
 }
```

Reply:
Yes, that makes sense, and I believe I have fixed it. Thanks!